Week1 jaehak

1TBhard opened the pull request:
week1 - Typing speed check

ChangHyun2 committed:
Rename read.md to README.md

ChangHyun2 committed:
Update README.md

1TBhard committed:
1 Week init repo

1TBhard committed:
[done] Type Speed Check

ChangHyun2 commented on jaehak/TypingSpeedCheck/main.js:
```diff
@@ -0,0 +1,145 @@
+		span.dataset.pressCorrect = "";
+		span.innerText = c;
+		return span;
+	});
```
```js

마크업을 리턴하고
sentenceToHTML(str){
return str.split('').map((c)=>`<span` class="character" pressCorrect="false">${c}</span>`).join('');
}

비동기함수에서 innerHTML로 elements를 추가해도 괜찮으려나요..?
loadNewSentence(){
  sentenceHTML= sentenceToMarkup(data);
  newSentence.innerHTML = sentenceHTML;
}

}
```

ChangHyun2 replied:
단어가 맞았는지 체크하는 부분과 관련해서 

comparisonStr = 'fetched str';
score = 0;
inputIndex= 0;
maxScore = comparisonStr.length;

input발생 시 inputIndex를 증가시키면서 
comparisonStr[inputIndex]와 e.data를 비교하고
맞을 경우 , score를 늘려서 채점 결과를 반환하는 방법도 있을 것 같아요
백스페이스를 입력할 경우 inputIndex를 감소시키고
inputIndex가 maxScore를 넘을 경우 게임이 종료되구요..?

돔 요소를 담는 배열을 순회하는 게 비용이 조금 크지 않을까 해서요..

첫 프로젝트부터 난이도가 높군요...ㅋㅋㅋㅋ
제가 짜지도 못할 코드 리뷰하는 것 같아서.. 조심스럽긴한데 틀린 부분은 전부 흘려보내주세요~!

1TBhard replied:
저도 부족한 부분이 많아서 많은 지적 부탁드립니다. ㅎㅎ

1TBhard replied:
마크업을 리턴하는 것이 더 깔끔하네요. 이 부분을 개선해야 겠군요!

score을 만들어서 채점결과를 반환하는 방법도 좋을 것 같네요. 

하지만, 이 프로젝트의 타이핑 속도를 채크하는 것에 목표를 두어서 score 보다는 타자 속도를 사용자에게 직접적으로 보여주는 것이 좋을 것 같네요. 

지적 감사드립니다.

1TBhard replied:
하나의 타이핑 테스트가 끝나면 결과가 나오도록 바꿨습니다.~

1TBhard replied:
마크업을 DOM 노드로 바꾸려고 노렸했는데 이를 해주는 함수가 없더군요..
`DOMParser.parseFromString()` 이라는 것이 있는데 이는 document 자체로 바꿔주네요..

일단 DOM 요소를 배열로 담고 해야겠습니다. 좀 더 좋은 방법을 찾아볼게요 ㅠㅠ

ChangHyun2 commented on jaehak/TypingSpeedCheck/main.js:
```diff
@@ -0,0 +1,145 @@
+  input.disabled = false;
+  input.onfocus;
+  loadNewSentence();
+  modalWrapper.classList.remove("active");
```
모달을 숨기는 부분은 reload handler로 넘기면 좋을 거 같아요~

```
document.getElementById("reload").addEventListener("click", ()=>{
modalWrapper.classList.remove('active');
startTyping();
});

```

1TBhard replied:
넵 리로드할 때로 같이 좋을 거 같네요. 수정했습니다.

1TBhard replied:
지금 보니까 아예 `startTyping()` 에 넣어서 시작시 한꺼번에 처리하는 것이 좋을 것 같아서 이로 수정했습니다.

ChangHyun2 commented on jaehak/TypingSpeedCheck/main.js:
```diff
@@ -0,0 +1,145 @@
+  myTimer = setInterval(() => {
+    document.getElementById("timer").innerText = `Time: ${++second}s`;
+    document.getElementById("speed").innerText = `Time: ${ Math.floor(progress.value / second) * 60}/min`;
+  }, 1000);
```
input-buffer에 keydown 핸들러를 등록해서
keydown이 발생할 때 timer를 실행하면 어떨까요?

modal이 사라졌을 때 runSpeedChecker를 실행하고

runSpeedChecker를
- loadNewSentence (곧바로 실행)
- startTyping (input-buffer 키 event 발생 시 실행)
- startTimer (input-buffer 키 event 발생 시 실행)

위처럼 나누면 좋을 거 같아요~!

그리고 loadNewSentence 메소드에는 $input-buffer.focus() 함수를 추가하면 좋을 거 같아요.
https://developer.mozilla.org/en-US/docs/Web/API/HTMLOrForeignElement/focus

1TBhard replied:
앗! 지금보니까 입력하지 않았는데도 timer 가 움직이는 건 사용자한테 가혹한 것 같네요.

바로 수정하겠습니다.

ChangHyun2 commented on jaehak/TypingSpeedCheck/readme.md:
```diff
@@ -0,0 +1,42 @@
+
+해당 이슈는 `addEventListener("input", )` 으로 처리하였다.
+
+- `input` : 입력되자 값이 `<input>`의 `value`에 바인딩 되고 이벤트 핸들러 실행
```
이 부분 내일 알려주심 감사하겟씁니다

1TBhard replied:
냅 알겠습니다.

ChangHyun2 reviewed (commented):
최선을 다했습니다...

1TBhard committed:
[Edit] Code review

1. 하나의 테스트 후 modal에 타이핑 속도(결과) 출력
2. 모달을 숨기는 부분을 reload handler이 아닌 startTyping() 에서 실행되게 함
3. input 태그에 키 눌러야 타이머 움직이기 시작

1TBhard commented on jaehak/TypingSpeedCheck/index.html:
```diff
@@ -1,50 +1,55 @@
+        </p>
+      </div>
+
+      <div id="result-speed"></div>
```
타이핑 속도 결과 `<div id="result-speed"></div>` 삽입

1TBhard committed:
[Edit] 1week Final

수정완료 - 재시작시 progress의 비정상 출력
수정완료 - 주어진 문장의 문자 사이 줄 간격이 생기는 것 수정(CSS)
